streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3302,6 +3304,76 @@ object ZStream extends ZStreamPlatformSpecificConstructors with Serializable {
   def fromJavaIteratorManaged[R, E, A](iterator: ZManaged[R, E, ju.Iterator[A]]): ZStream[R, E, A] =
     managed(iterator).flatMap(StreamEffect.fromJavaIterator)
 
+  /**
+   * Creates a managed (impure) Iterator from a Stream
+   * /!\ *Unsafe* for concurrent access. /!\
+   */
+  def toIterator[R, E, A](
```

Comment:
Why don't we just make this a method on the `ZStream` trait instead of the companion object so you can just do `stream.toIterator`? This would be analogous to the existing `toInputStream` method. There is also some good Scaladoc on that one about the iterator only being valid within the scope of the managed that we could borrow from.

Reply:
I didn't like the visibility of it, it is so impure. I will do it! 👍